### PR TITLE
Remove libm

### PR DESCRIPTION
**Pull request by Lokathor**
_Apparently_ you're not generally supposed to depend on and build the `libm` crate directly.

Instead, you should simply declare libm compatible `extern "C"` functions and call those. At link time LLVM will hook your code to (depending on platform) the system's version of `libm` (probably written in C) or, if that's not available, something from the "compiler-builtins" package, which is part of core, which uses the Rust `libm` crate internally as necessary.

The details of this process and the fact that it all works out are _not_ Stable in the Rust Stability sense, but it _does_ work like that and you end up with all code nicely using the same math functions across the whole program.

The _downside_ of all of this `extern "C"` stuff is that if you call a math function directly instead of using the `core::intrinsics` path to get there then LLVM will not perform any of its possible compile time optimizations such as const folding / evaluation. For example LLVM will recognize `core::intrinsics::sqrt(0.0)` and then compile time convert it to simply be `0.0` without actually calling the `sqrt` function. I'm not sure that this is a huge deal, but either way those intrinsics aren't stable and [the issue for that](https://github.com/rust-lang/rfcs/issues/2505) has been stagnant for a while.

Until `core` gets its act together we'll just do the "works without being guaranteed" thing. If later it stops working we'll fix it when that happens, probably be just going back to a direct dependency on the `libm` crate.

cc @gnzlbg: You don't need to check the content of the PR (though you can), but please read my above explanation of the situation to see if I got the story straight.

### CODE REVIEW

**Comment by Lokathor** [comment]
oh no it didn't work at all on like half the platforms we do CI with.

File: src/lib.rs [hunk 1]
```diff
@@ -409,7 +419,7 @@ pub fn sqrt_f32(f: f32) -> f32 {
   }
   #[cfg(not(target_feature = "sse"))]
   {
-    libm::sqrtf(f)
+    sqrtf(f)
```
Comment (gnzlbg) [inline src/lib.rs]: You are missing an `unsafe { sqrtf(f) }` here.

File: src/lib.rs [hunk 2]
```diff
@@ -428,6 +438,6 @@ pub fn sqrt_f64(f: f64) -> f64 {
   }
   #[cfg(not(target_feature = "sse2"))]
   {
-    libm::sqrt(f)
+    sqrt(f)
```
Comment (gnzlbg) [inline src/lib.rs]: and here.

File: src/lib.rs (hunk 1 above)
Comment (Lokathor) [reply src/lib.rs]: good eye

**Pull request closed without merging**